internal/provider/provider.go

```diff
@@ -47,15 +42,58 @@ type cacheKey struct {
 	method     string
 }
 
-func (p *provider) login(ctx context.Context, client *api.Client, vaultKubernetesMountPath, roleName, jwt string) (string, error) {
-	p.logger.Debug("vault: performing vault login...")
+func (p *provider) createJWTToken(ctx context.Context, podInfo config.PodInfo) (string, error) {
+	p.logger.Debug("creating service account token bound to pod",
+		"namespace", podInfo.Namespace,
+		"serviceAccountName", podInfo.ServiceAccountName,
+		"podName", podInfo.Name,
+		"podUID", podInfo.UID)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "", err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", err
+	}
+
+	ttl := int64((15 * time.Minute).Seconds())
+	resp, err := clientset.CoreV1().ServiceAccounts(podInfo.Namespace).CreateToken(ctx, podInfo.ServiceAccountName, &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			ExpirationSeconds: &ttl,
+			// TODO: Support audiences as a configurable.
+			//Audiences:         []string{},
+			BoundObjectRef: &authenticationv1.BoundObjectReference{
+				Kind:       "Pod",
+				APIVersion: "v1",
+				Name:       podInfo.Name,
+				UID:        podInfo.UID,
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to create a service account token for requesting pod %v: %w", podInfo, err)
```

Comment:
This will output the entire `podInfo` object. Would `podInfo.UID` be enough context in here?

Reply:
It would be if the pod still exists when the user reads the logs. But I left pod name + namespace in there in case the pod has since been deleted so a human reader can still understand some context. The service account name being logged is kind of just collateral but I figured does no harm.